sdk/java/pulumi/src/main/java/com/pulumi/test/PulumiTest.java

```diff
@@ -13,17 +16,68 @@
  */
 public interface PulumiTest {
 
+    /**
+     * Run a Pulumi test stack callback, wait for result and return {@link TestResult}.
+     *
+     * @param stack the stack to run in Pulumi test runtime
+     * @return a {@link TestResult} from Pulumi test runtime after running the stack
+     * @see #runTestAsync(Consumer)
+     * @see #withOptions(TestOptions)
+     * @see #withMocks(Mocks)
+     */
+    static TestResult runTest(Consumer<Context> stack) {
+        return withOptions(TestOptions.Empty).runTest(stack);
+    }
+
     /**
      * Run a Pulumi test stack callback asynchronously and return {@link TestResult}.
      *
      * @param stack the stack to run in Pulumi test runtime
      * @return a future {@link TestResult} from Pulumi test runtime after running the stack
+     * @see #runTest(Consumer)
+     * @see #withOptions(TestOptions)
+     * @see #withMocks(Mocks)
+     */
+    static CompletableFuture<TestResult> runTestAsync(Consumer<Context> stack) {
+        return withOptions(TestOptions.Empty).runTestAsync(stack);
+    }
+
+    /**
+     * Use {@link TestOptions} in this test.
+     *
+     * @param options the {@link TestOptions} to use
+     * @return a Pulumi test program entrypoint with given {@link TestOptions}
+     * @see #runTest(Consumer)
+     * @see #runTestAsync(Consumer)
+     * @see #withMocks(Mocks)
+     */
+    static PulumiTest.API withOptions(TestOptions options) {
+        return new PulumiTestInternal.APIInternal().withOptions(options);
+    }
+
+    /**
+     * Use {@link Mocks} in this test.
+     *
+     * @param mocks the {@link com.pulumi.deployment.internal.Monitor} mocks to use
+     * @return a Pulumi test program entrypoint with given {@link TestOptions}
+     * @see #runTest(Consumer)
+     * @see #runTestAsync(Consumer)
+     * @see #withOptions(TestOptions)
      */
-    CompletableFuture<TestResult> runTestAsync(Consumer<Context> stack);
+    static PulumiTest.API withMocks(Mocks mocks) {
+        return new PulumiTestInternal.APIInternal().withMocks(mocks);
+    }
 
+    /**
+     * Cleanup internal test state. <b>Must be called</b> after every run of a Pulumi test program.
+     */
+    static void cleanup() {
```

Comment:
This one is unfortunate. Resetting globals?

Reply:
yes, the globals :(